include/ghex/transport_layer/ucx/address_db_mpi.hpp

```diff
@@ -50,15 +50,14 @@ namespace gridtools {
                     key_t size() const noexcept { return m_size; }
                     int est_size() const noexcept { return m_size; }
 
-                    value_t* find(key_t k)
+                    value_t find(key_t k)
                     {
                         auto it = m_address_map.find(k);
                         if (it != m_address_map.end())
                         {
-                            return &(it->second);
+                            return it->second;
                         }
-                        else
-                            return nullptr;
+                        throw std::string("Cound not find peer address in the MPI address xdatabase.");
```

Comment:
please throw a runtime_error and not a simple string

Reply:
done